packages/jsii-reflect/lib/type-system.ts

```diff
@@ -26,6 +27,31 @@ export class TypeSystem {
 
   private readonly _assemblyLookup: { [name: string]: Assembly } = { };
 
+  /**
+   * Load all JSII dependencies of the given NPM package directory.
+   *
+   * The NPM package itself does *not* have to be a jsii package, and does
+   * NOT have to declare a JSII dependency on any of the packages.
+   */
+  public async loadAllNodeModules(packageRoot: string, options: { validate?: boolean } = {}) {
```

Comment:
Would make more sense (to me) to look at `package.json` and just load all modules from it (through `require.resolve`). Technically, there could be modules in node_modules that reside on parent directories.

Reply:
Good point.